tests/appsec/test_automated_login_events.py

```diff
@@ -1348,3 +1218,761 @@ def validate_iden(meta):
+                assert meta["appsec.events.users.login.failure.usr.id"] == USER_HASH
+                assert meta["_dd.appsec.usr.id"] == USER_HASH
+
+    def setup_login_sdk_success_local(self):
+        self.r_sdk_success = weblog.post(
+            "/login?auth=local&sdk_event=success&sdk_user=sdkUser",
+            data=login_data(context, USER, PASSWORD),
+        )
+
+    def test_login_sdk_success_local(self):
+        assert self.r_sdk_success.status_code == 200
+        for _, trace, span in interfaces.library.get_spans(request=self.r_sdk_success):
+            assert_priority(span, trace)
+            meta = span.get("meta", {})
+
+            # mandatory
+            assert meta["appsec.events.users.login.success.usr.login"] == "sdkUser"
+            assert meta["_dd.appsec.usr.login"] == USERNAME_HASH
+            assert meta["_dd.appsec.events.users.login.success.auto.mode"] == "anonymization"
+            assert meta["appsec.events.users.login.success.track"] == "true"
+            assert meta["_dd.appsec.events.users.login.success.sdk"] == "true"
+
+            # optional (to review for each library)
+            if context.library not in libs_without_user_id:
+                assert meta["usr.id"] == "sdkUser"
+                assert meta["_dd.appsec.usr.id"] == USER_HASH
+
+    def setup_login_sdk_success_basic(self):
+        self.r_sdk_success = weblog.get(
+            "/login?auth=basic&sdk_event=success&sdk_user=sdkUser",
+            headers={"Authorization": BASIC_AUTH_USER_HEADER},
+        )
+
+    @missing_feature(context.library == "php", reason="Basic auth not implemented")
+    def test_login_sdk_success_basic(self):
+        assert self.r_sdk_success.status_code == 200
+        for _, trace, span in interfaces.library.get_spans(request=self.r_sdk_success):
+            assert_priority(span, trace)
+            meta = span.get("meta", {})
+
+            # mandatory
+            assert meta["appsec.events.users.login.success.usr.login"] == "sdkUser"
+            assert meta["_dd.appsec.usr.login"] == USERNAME_HASH
+            assert meta["_dd.appsec.events.users.login.success.auto.mode"] == "anonymization"
+            assert meta["appsec.events.users.login.success.track"] == "true"
+            assert meta["_dd.appsec.events.users.login.success.sdk"] == "true"
+
+            # optional (to review for each library)
+            if context.library not in libs_without_user_id:
+                assert meta["usr.id"] == "sdkUser"
+                assert meta["_dd.appsec.usr.id"] == USER_HASH
+
+    def setup_login_sdk_failure_basic(self):
+        self.r_sdk_failure = weblog.get(
+            "/login?auth=basic&sdk_event=failure&sdk_user=sdkUser&sdk_user_exists=true",
+            headers={"Authorization": BASIC_AUTH_INVALID_USER_HEADER},
+        )
+
+    @missing_feature(context.library == "php", reason="Basic auth not implemented")
+    def test_login_sdk_failure_basic(self):
```

Comment:
for some reason the `test_login_sdk_failure_basic` is right before the `test_login_sdk_failure_local` even tho everywhere else, the local test comes before the basic test. very small nit, but hey, why not keep it consistent

Reply:
I've just reordered it, thanks!